services/postService.js

```diff
@@ -0,0 +1,39 @@
+import Post from '../models/Post.js';
+import Hashtag from '../models/Hashtag.js';
+
+export const create = async (data) => {
+  const post = new Post(data);
+  const hashtags = data.content.match(/#[^\s#]+/g);
+  const result = await Promise.all(
+    hashtags.map(async tagWithHash => {
+      const tag = tagWithHash.slice(1).toLowerCase();
+      const r = await Hashtag.findOne({ tag }) || await Hashtag.create({ tag });
+      return r;
+    })
+  );
```

Comment:
```suggestion
const addOrFindHashTags = hashtags.map(async tagWithHash => {
   const tag = tagWithHash.slice(1).toLowerCase();
   const r = await Hashtag.findOne({ tag }) || await Hashtag.create({ tag });
   return r;
 });
 
const result = await Promise.all(addOrFindHashTags);
```

요럼 좀 더 직관적으로 보일거 같아요!

Reply:
@mango906 @sohyeong-dev 해쉬태그 하나 하나 findOne, create 쿼리를 날려 주면 IO 부하가 많이 일어나지 않을까요~?

`Model.insertMany()` 으로 한 번의 통신으로 넣어 보는 건 어떨까요??

Reply:
어 근데 저 쿼리의 목적이 해당 해쉬태그가 있으면 그걸 가져오고 없으면 새로 만드는거 아닌가욥??

Reply:
@mango906 그러면 해쉬태그 전체 삽입하고, 전체 가져오는, 단 두 개의 쿼리만으로도 되지 않을까?!